src/v/kafka/latency_probe.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+#include "utils/hdr_hist.h"
+
+#include <seastar/core/metrics.hh>
+
+namespace kafka {
+class latency_probe {
+public:
+    void setup_metrics() {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_metrics()) {
+            return;
+        }
+        std::vector<sm::label_instance> labels{sm::label("latency")("measure")};
+        _metrics.add_group(
+          prometheus_sanitize::metrics_name("fetch_and_produce_latency_ms"),
+          {sm::make_histogram(
+             "fetch_latency_ms",
+             sm::description("Fetch Latency"),
+             labels,
+             [this] { return _fetch_latency.seastar_histogram_logform(); }),
+           sm::make_histogram(
+             "produce_latency_ms",
+             sm::description("Produce Latency"),
+             labels,
+             [this] { return _produce_latency.seastar_histogram_logform(); })});
+    }
+
+    std::unique_ptr<hdr_hist::measurement> auto_produce_measurement() {
+        return _produce_latency.auto_measure();
+    }
+    std::unique_ptr<hdr_hist::measurement> auto_fetch_measurement() {
+        return _fetch_latency.auto_measure();
+    }
+
+private:
+    hdr_hist _produce_latency;
```

Comment:
this needs to change like I mentioned in the other ticket. you need to have a new constructor that reduces memory footprint.